### ISSUE TITLE
Complex Match Queries

### ISSUE THREAD

**Issue by ksgnextuple** [Telefonica/prometheus-kafka-adapter#133]
Hi 

Is it possible to have such a MATCH expression ['histogram_quantile(\"0.95\", sum by(le) (rate(http_server_requests_seconds_bucket{uservice=\"<something>\", uri=\"<something>\",kubernetes_namespace=\"<something>\"}[5m])))']?

**Comment by palmerabollo** [comment]
I don't think so. The parser uses [expfmt](https://pkg.go.dev/github.com/prometheus/common/expfmt) to validate the match expression https://github.com/Telefonica/prometheus-kafka-adapter/blob/702387c65b91cdefc35dc2eac1f294db821aa908/config.go#L154. That part could work. But the logic that filters the metrics https://github.com/Telefonica/prometheus-kafka-adapter/blob/702387c65b91cdefc35dc2eac1f294db821aa908/serializers.go#L126 only uses name & label, so I don't think other functions such as `histogram_quantile` would work.

@Gallardot implemented that part in https://github.com/Telefonica/prometheus-kafka-adapter/pull/64, perhaps he can confirm.

**Comment by Gallardot** [comment]
> I don't think so. The parser uses [expfmt](https://pkg.go.dev/github.com/prometheus/common/expfmt) to validate the match expression https://github.com/Telefonica/prometheus-kafka-adapter/blob/702387c65b91cdefc35dc2eac1f294db821aa908/config.go#L154. That part could work. But the logic that filters the metrics https://github.com/Telefonica/prometheus-kafka-adapter/blob/702387c65b91cdefc35dc2eac1f294db821aa908/serializers.go#L126 only uses name & label, so I don't think other functions such as `histogram_quantile` would work.
> 
> @Gallardot implemented that part in https://github.com/Telefonica/prometheus-kafka-adapter/pull/64, perhaps he can confirm.

@palmerabollo Agree with you.


Hi @ksgnextuple,

This expression looks more like a Prometheus query expression. However, the prometheus-kafka-adapter itself does not have data storage capabilities, so I believe this requirement is not easily achievable. If you have more context or additional requirements, please let me know.

Thank you!
